hedwig/backends/gcp.py

```diff
@@ -196,6 +196,16 @@ def shutdown(self, await_msg_callbacks=False) -> None:
         # instead, it's faster to actually process all the messages
 
 
+class ShutdownMessageScheduler(PubSubMessageScheduler):
+    def __init__(self, work_queue: Queue, subscription_path: str, shutdown_event: threading.Event):
+        super().__init__(work_queue, subscription_path)
+        self.shutdown_event = shutdown_event
+
+    def shutdown(self, await_msg_callbacks=False) -> List:  # type: ignore
+        self.shutdown_event.set()
```

Comment:
hm not sure how I feel about this - the event is technically just an input to this class and owned by the caller. I'd rather we raise an exception here and let caller deal with that instead of this - thoughts?

Reply:
Yeah, I dont think we need this shutdown event here